packages/flutter/test/widgets/nested_scroll_view_test.dart

```diff
@@ -109,7 +109,139 @@ Widget buildTest({
   );
 }
 
+
+class _TabBarViewInNestedScrollView extends StatefulWidget {
+  const _TabBarViewInNestedScrollView();
+
+  @override
+  State<_TabBarViewInNestedScrollView> createState() =>
+      _TabBarViewInNestedScrollViewState();
+}
+
+class _TabBarViewInNestedScrollViewState
+    extends State<_TabBarViewInNestedScrollView> with TickerProviderStateMixin {
+  final List<int> _tabs = <int>[0, 1];
+
+  late TabController _tabController;
+  int _selectedIndex = 0;
+
+  @override
+  void initState() {
+    super.initState();
+    _tabController = TabController(
+      initialIndex: _selectedIndex,
+      vsync: this,
+      length: _tabs.length,
+    );
+    _tabController.addListener(() {
+      setState(() {
+        _selectedIndex = _tabController.index;
+      });
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: NestedScrollView(
+        floatHeaderSlivers: true,
+        headerSliverBuilder: (BuildContext context, bool innerBoxIsScrolled) {
+          return <Widget>[
+            SliverAppBar(
+              title: const Text('title'),
+              pinned: true,
+              expandedHeight: 150.0,
+              bottom: TabBar(
+                controller: _tabController,
+                tabs:
+                _tabs.map((int index) => Tab(text: 'tab $index')).toList(),
+              ),
+            ),
+          ];
+        },
+        body: TabBarView(
+          controller: _tabController,
+          children: _tabs
+              .map((int e) => _TabPage(
+            tab: e,
+            isSelected: _selectedIndex == e,
+          ))
+              .toList(),
```

Comment:
I feel like this could be modified for improved readability, for example:

```suggestion
          children: <Widget>[
            for (final int index in _tabs)
              _TabPage(tab: index, isSelected: _selectedIndex == index),
          ],
```

Reply:
👍 
I improved the readability in b4197b0